exsim/ltem.py

```diff
@@ -0,0 +1,92 @@
+import numpy as np
+import discretisedfield as df
+import micromagneticmodel as mm
+from scipy import constants
+
+
+# TODO
+# - remove computation of wavelength (not required)
+# - recheck that all units are correct, everything should be in SI units
+# - if possible: remove transposing before/after Fourier transform
+# - Write a proper documentation.
+# - Cs is not used. Can it be removed?
+# - Write tests.
+def ltem_phase(field, /, U, Cs, kx=0.1, ky=0.1):
+    """LTEM phase contrast.
+
+    Parameters
+    ----------
+    field : df.Field
+        Magnetisation field.
+    U : numbers.Real
+        Accelerating voltage of electrons in V.
+    Cs : numbers.Real
+        Spherical aberration coefficient
+    kx : numbers.Real, optional
+        Tikhonov filter radius in x in pixels.
+    ky : numbers.Real, optional
+        Tikhonov filter radius in y in pixels.
+
+    Returns
+    -------
+    df.Field
+        Phase in real space.
+    df.Field
+        Phase in Fourier space.
+
+    Example
+    -------
+    ...
+    """
+    const = 1j * field.mesh.region.edges[2] / (
+        2 * constants.codata.value('mag. flux quantum'))
+    m_sat = np.max(field.norm.array) * mm.consts.mu0
+
+    wavelength = _relativistic_wavelength(U)
+    print(f'The electron beam has a wavelength of {wavelength * 1e9:.2e} nm.')
+
+    mx_projection = np.transpose(field.orientation.x.project('z').array.squeeze())
+    my_projection = np.transpose(field.orientation.y.project('z').array.squeeze())
+
+    ft_mx = np.fft.fft2(mx_projection, axes=(-2, -1))
+    ft_my = np.fft.fft2(my_projection, axes=(-2, -1))
```

Comment:
FFT is something we should definitely have in `df`. Maybe when we gather more information about FFTs in other `exsim` tools, we can decide what the best way to implement it would be.

Reply:
If we think about `fft = field.fft()` we can maybe:

- consider `fft` being a `Field` object
- the mesh of that field having the coordinates computed using `np.fft.fftfreq`

That would simplify the next few lines a lot.

Reply:
Something like this would be good. As I understand it this might also help in removing `transpose`.